internal/testutils/stats/test_metrics_recorder.go

```diff
@@ -180,3 +180,19 @@ func (r *TestMetricsRecorder) TagConn(ctx context.Context, _ *stats.ConnTagInfo)
 }
 
 func (r *TestMetricsRecorder) HandleConn(context.Context, stats.ConnStats) {}
+
+// NoopMetricsRecorder is a noop MetricsRecorder to be used in tests to prevent
+// nil panics.
+type NoopMetricsRecorder struct{}
+
+func (r *NoopMetricsRecorder) RecordInt64Count(_ *estats.Int64CountHandle, _ int64, _ ...string) {}
+
+func (r *NoopMetricsRecorder) RecordFloat64Count(_ *estats.Float64CountHandle, _ float64, _ ...string) {
```

Comment:
`gofmt` is crazy that it allows `{` and `}` to be on the same line for some of the methods and not for other methods. I've noticed this when making changes too. 

You could get rid of the `_` parameter names for all the methods though, since none of the parameters are used.

Reply:
Ah I see. Will switch to one line, and remove _ parameter name.